invenio_circulation/search/api.py

```diff
@@ -60,12 +60,17 @@ def search_by_pid(item_pid=None, document_pid=None, filter_states=None,
     return search
 
 
-def search_by_patron_item(patron_pid, item_pid, filter_states=None):
+def search_by_patron_item_or_document(patron_pid, item_pid=None,
+                                      document_pid=None, filter_states=None):
     """Retrieve loans for patron given an item."""
     search = current_circulation.loan_search
     search = search \
-        .filter("term", patron_pid=patron_pid) \
-        .filter("term", item_pid=item_pid)
+        .filter("term", patron_pid=patron_pid)
+
+    if item_pid:
+        search = search.filter("term", item_pid=item_pid)
+    else:
```

Comment:
I would say that you can omit this else and put an if so we can search by both. WDYT? Also document_pid can be none so a check would be good I think.